src/main/java/gov/cabinetoffice/gap/adminbackend/services/ApplicationFormSectionService.java

```diff
@@ -118,4 +118,35 @@ public void updateSectionStatus(final Integer applicationId, final String sectio
         this.applicationFormRepository.save(applicationForm);
     }
 
+    public void updateSectionOrder(final Integer applicationId, final String sectionId, final Integer increment) {
+        AdminSession session = HelperUtils.getAdminSessionForAuthenticatedUser();
+
+        ApplicationFormEntity applicationForm = this.applicationFormRepository.findById(applicationId)
+                .orElseThrow(() -> new NotFoundException("Application with id " + applicationId + " does not exist"));
+
+        if (!session.getGrantAdminId().equals(applicationForm.getCreatedBy())) {
+            throw new AccessDeniedException("User " + session.getGrantAdminId()
+                    + " is unable to access the application form with id " + applicationId);
+        }
+
+        List<ApplicationFormSectionDTO> sections = applicationForm.getDefinition().getSections();
+        ApplicationFormSectionDTO section = applicationForm.getDefinition().getSectionById(sectionId);
+        int index = sections.indexOf(section);
+
+        final int BOTTOM_ENTRY = sections.size() - 1;
```

Comment:
`BOTTOM_ENTRY` is a bit confusing. `SECTION_LIST_SIZE` might be more understandable